exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/HashPartition.java

```diff
@@ -157,11 +162,11 @@ public HashPartition(FragmentContext context, BufferAllocator allocator, Chained
         .build(logger);
     } catch (SchemaChangeException sce) {
       throw new IllegalStateException("Unexpected Schema Change while creating a hash table",sce);
-    }
-    this.hjHelper = semiJoin ? null : new HashJoinHelper(context, allocator);
-    tmpBatchesList = new ArrayList<>();
-    if (numPartitions > 1) {
-      allocateNewCurrentBatchAndHV();
+    } catch (OutOfMemoryException oom) {
+      close();
+      throw UserException.memoryError(oom)
+          .message("OutOfMemory while allocate memory for hash partition.")
```

Comment:
Suggested: `"Failed to allocate hash partition."`

The `memoryError()` already indicate it is an OOM error.